tests_integration/test_sql.py

```diff
@@ -49,3 +55,36 @@ def _round_values(results):
 
 def _round_values_for_multi_queries(results):
     return [_round_values(r) for r in results]
+
+
+@pytest.mark.integration
+@mock.patch("snowcli.output.printing._get_table")
+@mock.patch("snowcli.output.printing.Live")
+def test_queries_are_streamed_to_output(
+    _, mock_get_table, runner, capsys, test_root_path
+):
+    log = []
+
+    def _log(
+        *args,
+    ):
+        log.append((time.monotonic(), *args))
+
+    # We mock add_row by a function that logs call time, later we compare call times
+    # to make sure those were separated in time
+    mock_get_table().add_row = _log
+
+    runner.invoke_with_connection(
+        [
+            "sql",
+            "-q",
+            "select 13; select system$wait(10);",
+        ],
+    )
+
+    assert len(log) == 2
+    (time_0, query_0), (time_1, query_1) = log
+
+    assert query_0 == "13"
+    assert time_1 - time_0 >= 10.0
```

Comment:
```suggestion
    assert time_1 - time_0 >= 7.5
```

IMO setting the same exact value as in `wait` can cause flaky tests. If the first response from Snowflake will be slower than the second one, then this test may fail.

Example:
a. Time 0.0: Request sent to Snowflake
b. Time 0.1:  Request received in Snowflake
c. Time 0.2: Response prepared in Snowflake and transfer started to CLI
d. Time 0.2: `Wait(10)` started
e. Time 0.7: Response received in SnowCLI (time in `log`: 0.7), response transfer took 0.5
f. Time 10.2: `Wait(10)` completed and transfer of response to CLI started
g. Time 10.5: Response received in SnowCLI (time in `log`: 10.5), response transfer took 0.3

```
g - e = 10.5 - 0.7 = 9.8
9.8 < 10
assertion fails
```

Reply:
One for the advantages of proposed solution is that ignores latency. We just check time of printing. Due to using iterator executing of wait query will occur only after printing results of first query (calling next). It means that previous latency does not matter. In fact we have a guarantee that the time will be bigger than 10 seconds:
t=0 print results of query 1
t=0.1 execute wait query
t=10.2 print wait query

Reply:
Ok, I thought that `execute_stream` sends all queries together and only reads responses as a stream. I didn't look at its implementation. Now I see what you mean and you're right - it will work correctly.